### PR TITLE
[LC-55] Block Versioning System

### PR DESCRIPTION
**Pull request by zsaladin**

### CODE REVIEW

File: cli_tools/icx_test/icx_wallet.py [hunk 1]
```diff
@@ -19,7 +19,9 @@ def __init__(self, private_key=None):
         self.__private_key = private_key or PrivateKey()
         self.__address = self.create_address(self.__private_key.pubkey)
         self.__last_tx_hash = ""
-        self.__hash_generator = get_tx_hash_generator(conf.LOOPCHAIN_DEFAULT_CHANNEL)
+
+        tx_hash_version = conf.CHANNEL_OPTION[conf.conf.LOOPCHAIN_DEFAULT_CHANNEL]["tx_hash_version"]
```
Comment (zsaladin) [inline cli_tools/icx_test/icx_wallet.py]: `conf.conf` mis typos

File: loopchain/peer/consensus_base.py [hunk 2]
```diff
@@ -23,81 +20,51 @@ class ConsensusBase(metaclass=ABCMeta):
...
+            except Exception as e:
+                logging.warning(f"tx hash invalid. tx: {tx}")
 
-        if peer_manager_block is not None:
-            logging.debug("consensus_base:_makeup_block make peer_list block and sign it.")
-            peer_manager_block.generate_block(self._candidate_blocks.get_last_block(self._blockchain))
-            peer_manager_block.sign(
-                ObjectManager().channel_service.peer_auth
-            )
+            block_builder.transactions[tx.hash] = tx
```
Comment (zsaladin) [inline loopchain/peer/consensus_base.py]: else: needed

File: loopchain/baseservice/broadcast_scheduler.py [hunk 3]
```diff
@@ -188,11 +182,12 @@ def __call_async_to_target(self, peer_target, method_name, method_param, is_stub
         except KeyError as e:
             logging.debug(f"broadcast_thread:__call_async_to_target ({peer_target}) not in audience. ({e})")
         else:
-            stub_item.call_async(method_name=method_name,
-                                 message=method_param,
-                                 is_stub_reuse=is_stub_reuse,
-                                 call_back=call_back_partial,
-                                 timeout=timeout)
+            if stub_item:
```
[the remark this answers is not in the document]
Comment (zsaladin) [reply loopchain/baseservice/broadcast_scheduler.py]: It is not a part of if statement but try except.

File: loopchain/peer/consensus_siever.py [hunk 4]
```diff
@@ -23,144 +26,97 @@ class ConsensusSiever(ConsensusBase):
...
-                self._blockmanager.broadcast_send_unconfirmed_block(self._block)
-
-                self._stop_gen_block()
-                util.logger.spam(f"consensus_siever:consensus channel({self._channel_name}) "
-                                 f"\ntry ObjectManager().peer_service.rotate_next_leader({self._channel_name})")
-
-                ObjectManager().channel_service.state_machine.turn_to_peer()
-
-        return result
+        block_builder = self._makeup_block()
```
Comment (yakkle) [inline loopchain/peer/consensus_siever.py]: block_builder is shadow name from `loopchain/blockchain/blocks/block_builder`.
ref : https://stackoverflow.com/q/20125172

Comment (zsaladin) [reply loopchain/peer/consensus_siever.py]: Fixed it by modifying import packages.

**Review by windies21 — APPROVED** [review]

**Pull request merged**